src/dependencies/useStoreDependency.ts

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import {
+  Dependency,
+  calculate,
+  calculateForDispatch,
+  makeDependencyIndex,
+  DependencyIndexEntry,
+} from './DependencyMap';
+import { Dispatcher } from 'flux';
+import { get as getDispatcherInstance } from '../dispatcher/DispatcherInstance';
+import { enforceDispatcher } from '../dispatcher/DispatcherInterface';
+import { handleDispatch } from './Dispatch';
+
+type SingleDependency = {
+  [key: string]: Dependency;
+};
+
+function useStoreDependency<Props>(
+  dependency: Dependency,
+  props: Props,
+  dispatcher: Dispatcher<any> = getDispatcherInstance()
+) {
+  enforceDispatcher(dispatcher);
+
+  const dispatchToken = React.useRef(null);
+  const { current: dependencyMap } = React.useRef<SingleDependency>({
+    dependency,
+  });
+  const { current: dependencyIndex } = React.useRef(
+    makeDependencyIndex(dependencyMap)
```

Comment:
Are you trying to calculate `dependencyIndex` only once? I wonder if this should be `useState`

Reply:
yeah we need a single `dependencyIndex` that's going to stay consistent across renders

Reply:
This should work and have the advantage of only "initializing" once
```es6
const [dependencyIndex] = useState(() => makeDependencyIndex(dependencyMap))`
```

Reply:
this should work effectively the same, no? i thought the initial value to `useRef` and `useState` function the same where it's only evaluated once. otherwise the ref's value would get overwritten each render

Reply:
true. I guess I meant that the `useState` version would only call `makeDependencyIndex()` once.

Reply:
oh i see. the ref version will call it every time even if it's not used, whereas the state version will only execute it once